### ISSUE TITLE
A11y: [Chats] Keyboard focus trapped in search view; Close butto...

### ISSUE THREAD

**Issue by mxandreas** [element-hq/element-x-android#6400]
- **Checkpoint ID:** 11.2.4.3/4.2
- **Component:** Chats

### Issue Description
Keyboard focus trapped in search view; Close button doesn't return focus.

### Recommendation
Fix focus trap and ensure focus returns to '+' button on close.

--- 
*Report Date: 25.02.2026*

**Comment by bmarty** [comment]
For the first point:

> When you open the “Start chat” screen via the “+” button and then close it using the Close
button, the keyboard focus does not return to the “+” button. Instead, focus is moved to the
“Settings” button at the top of the page. This forces keyboard users to navigate through all
controls again to get back to where they were. Focus should be restored to the “+” button.

This is a global issue in all the application. I believe that the mono-activity architecture is the source of the issue. The keyboard focus, as well as the activity focus is lost when something is added to the Appyx backstack. Maybe this will be fixed when we will migrate to navigation3.

For the second point:

> There is an additional issue: after opening “Start chat” via the “+” button, moving focus to
the “Search for someone” input, and then trying to close the screen using the Close button,
this does not work. Focus is automatically moved back to the “Search for someone” input
field, effectively trapping keyboard focus. The search view can only be closed using the
Escape key.

Hitting the `Tab` key on the external keyboard allow to navigate out of the search field. So there is no issue to fix.

I move the issue to the blocked state for the first point.

**Comment by bmarty** [comment]
@mxandreas should we close this issue?

**Comment by mxandreas** [comment]
> [@mxandreas](https://github.com/mxandreas) should we close this issue?

Yes, I think so.

**Comment by bmarty** [comment]
For posterity, I have checked the behavior of the Navigation3 library (using code in https://github.com/element-hq/element-x-android/pull/7179) and the behavior is the same: the keyboard focus is not restored at the same position when navigating back to the file list:

https://github.com/user-attachments/assets/803835c0-b838-45d8-9333-82d9336a46ed
